src/Integration/Persistence/FileBindingSerializer.cs

```diff
@@ -0,0 +1,230 @@
+            this.fileWrapper = fileWrapper;
+        }
+
+        protected abstract WriteMode Mode { get; }
+
+        protected abstract string GetFullConfigurationFilePath();
+
+        protected virtual bool OnSuccessfulFileWrite(string filePath)
+        {
+            // Default implementation is a no-op
+            return true;
+        }
+
+        internal ICredentialStore Store
+        {
+            get { return this.credentialStore; }
+        }
+
+        public BoundSonarQubeProject ReadSolutionBinding()
+        {
+            string configFile = this.GetFullConfigurationFilePath();
+            if (string.IsNullOrWhiteSpace(configFile) || !fileWrapper.Exists(configFile))
+            {
+                return null;
+            }
+
+            return this.ReadBindingInformation(configFile);
+        }
+
+        public string WriteSolutionBinding(BoundSonarQubeProject binding)
+        {
+            if (binding == null)
+            {
+                throw new ArgumentNullException(nameof(binding));
+            }
+
+            string configFile = this.GetFullConfigurationFilePath();
+            if (string.IsNullOrWhiteSpace(configFile))
+            {
+                return null;
+            }
+
+            sccFileSystem.QueueFileWrite(configFile, () =>
+            {
+                if (this.WriteBindingInformation(configFile, binding))
+                {
+                    return OnSuccessfulFileWrite(configFile);
+                }
+
+                return false;
+            });
+
+            if (this.Mode == WriteMode.Immediate && !sccFileSystem.WriteQueuedFiles())
+            {
+                return null; // write operation failed
+            }
+
+            return configFile;
+        }
+        
+        private BoundSonarQubeProject ReadBindingInformation(string configFile)
+        {
+            BoundSonarQubeProject bound = this.SafeDeserializeConfigFile(configFile);
+            if (bound?.ServerUri != null)
+            {
+                var credentials = this.credentialStore.ReadCredentials(bound.ServerUri);
+                if (credentials != null)
+                {
+                    bound.Credentials = new BasicAuthCredentials(credentials.Username,
+                        credentials.Password.ToSecureString());
+                }
+            }
+
+            return bound;
+        }
+
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Reliability",
+            "S3215:\"interface\" instances should not be cast to concrete types",
+            Justification = "Casting as BasicAuthCredentials is because it's the only credential type we support. Once we add more we need to think again on how to refactor the code to avoid this",
+            Scope = "member",
+            Target = "~M:SonarLint.VisualStudio.Integration.Persistence.FileBindingSerializer.WriteBindingInformation(System.String,SonarLint.VisualStudio.Integration.Persistence.BoundProject)~System.Boolean")]
```

Comment:
I updated the fully qualified name in the suppression target (forgot to do in the change I made yesterday)